collector/src/tests/Makefile

```diff
@@ -14,7 +14,7 @@ else
 CXXFLAGS += -g -O2     # release mode; NOTE: without -g the creation of debuginfo RPMs will fail in COPR!
 LDFLAGS += -g -O2
 endif
-LIBS += -lfmt -lgtest -lpthread
+LIBS += -lfmt -lgtest -lpthread -lprometheus-cpp-pull -lprometheus-cpp-core -lz
```

Comment:
let's define a GNU make variable "PROMETHEUS_SUPPORT" which can be 0 or 1.

ifeq ($(PROMETHEUS_SUPPORT),1)
LIBS += -lprometheus-cpp-pull -lprometheus-cpp-core -lz
CXXFLAGS += -DPROMETHEUS_SUPPORT=1
endif